src/App.tsx

```diff
@@ -1,7 +1,30 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+export class App extends React.Component {
+  state = {
+    lastKey: '',
+    listener: (event: KeyboardEvent) => {
+      this.setState({ lastKey: event.key });
+    },
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', this.state.listener);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.state.listener);
+  }
+
+  render() {
+    return (
+      <div
+        className="App"
+      >
+        {this.state.lastKey
+          ? <p className="App__message">{`The last pressed key is [${this.state.lastKey}]`}</p>
+          : <p className="App__message">Nothing was pressed yet</p>}
+      </div>
```

Comment:
you're rendering `<p>` anyway right?
so why not just conditionally set the contents inside <p> instead of repeating the <p> twice?
```suggestion
      <p className="App__message">
        {this.state.lastKey
          ? `The last pressed key is [${this.state.lastKey}]`
          : 'Nothing was pressed yet'}
        </p>
      </div>
```